05/ContactForm.js

```diff
@@ -1,10 +1,110 @@
-import React from 'react';
+import React, { useReducer, useState } from 'react';
 
-import account from './account';
+// import account from './account';
 
-const ContactForm = () => {
-    console.log(account);
-    return <form>??</form>;
+const ContactForm = function () {
+    const init = { firstNameAndLastName: '', email: '', phoneNumber: '', topic: '', message: '' };
+    const reducer = (state, { name, value }) => {
+        const newState = { ...state };
+        newState[name] = value;
+        return newState;
+    };
+
+    const [state, dispatch] = useReducer(reducer, init);
+    const { firstNameAndLastName, email, phoneNumber, topic, message } = state;
+    const [error, setError] = useState(null);
+    const fields = [
+        {
+            name: 'firstNameAndLastName',
+            label: 'firstNameAndLastName',
+            required: true,
+            pattern: '^[a-zA-Z-zżźćńółęąśŻŹĆĄŚĘŁÓŃ]+ [a-zA-Z-zżźćńółęąśŻŹĆĄŚĘŁÓŃ]+$',
+        },
+        {
+            name: 'email',
+            label: 'email',
+            required: true,
+            pattern: '([a-zA-Z0-9\\_\\-\\.]+)@([a-zA-Z]+).(.+)',
+        },
+        {
+            name: 'phoneNumber',
+            label: 'phoneNumber',
+            required: false,
+            pattern: '^[0-9]{9}$',
+        },
+        {
+            name: 'topic',
+            label: 'topic',
+            required: true,
+        },
+        {
+            name: 'message',
+            label: 'message',
+            required: true,
+        },
+    ];
+
+    const validate = () => {
+        let err = 'Formularz wypełniony prawidłowo!';
```

Comment:
Czy to jest błąd? Raczej nie :p

PS. Może warto przechowywać tablicę, aby zebrać od razu wszystkie błędy, a nie robić to "od ostatniego"